Sources/PortalSwift/Provider/PortalProvider.swift

```diff
@@ -214,82 +177,255 @@ public class PortalProvider {
 
   /// Makes a request.
   /// - Parameters:
-  ///   - payload: A normal payload whose params are of type [Any].
-  ///   - completion: Resolves with a Result.
-  /// - Returns: Void
+  ///   - chainId: A CAIP-2 Blockchain ID associated with the request.
+  ///   - withMethod: A member of the PortalRequestMethod enum
+  ///   - andParams: An array of parameters for the request (either RPC parameters or a transaction if signing)
+  /// - Returns: PortalProviderResult
```

Comment:
A new `async` request method that takes any `Encodable` param as an argument. Allows for type flexibility for customers.